#!/usr/bin/env python
PACKAGE = "franka_motion_primitive"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# (name, type, level, description, dfault, min, max)
gen.add("kd_lin", double_t, 0, "vd = -kd*f_sense", 0.01, 0, 1.)
gen.add("kd_rot", double_t, 0, "vd = -kd*f_sense", 0.1, 0, 500.)
gen.add("stiffness_lin", double_t, 0, "vd = controller stiffness", 1000, 0, 3000)
gen.add("stiffness_rot", double_t, 0, "vd = controller stiffness", 50, 0, 120)
gen.add("damping_lin", double_t, 0, "controller damping", 70, 0, 110)
gen.add("damping_rot", double_t, 0, "controller damping", 14, 0, 22)

exit(gen.generate(PACKAGE, "dynamic_admittance_param", "Admittance"))
